### PR TITLE
feat(evm-state): squash evm state

### PR DESCRIPTION
**Pull request by hrls**

### CODE REVIEW

File: evm-utils/evm-state/src/layered_backend.rs [hunk 1]
```diff
@@ -189,9 +189,39 @@ impl EvmState {
             "new slot {} with previous {:?}",
             self.current_slot, self.previous_slot
         );
+
         self.storage
             .new_version(self.current_slot, self.previous_slot)
             .expect("Unable to create new version in storage");
+
+        const SQUASH_THRESHOLD: u64 = 100; // TODO: avoid storage reads
```
Comment (vldm) [inline evm-utils/evm-state/src/layered_backend.rs]: Please remove constant, and use squash explicit in snapshot method.

**Review by vldm — APPROVED** [review]
Rebase please

File: evm-utils/evm-state/src/layered_backend.rs (hunk 1 above)
Comment (hrls) [reply evm-utils/evm-state/src/layered_backend.rs]: Probably this call should be as part of bank squash behaviour. Snapshot can be disabled actually.

File: evm-utils/evm-state/benches/squashed_state.rs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+};
+
+const N_ACCOUNTS: usize = 1024;
+const AVERAGE_DATA_SIZE: usize = 2 * 1024;
+
+fn some_account() -> AccountState {
+    AccountState {
+        nonce: U256::from(random::<u64>()),
+        balance: U256::from(random::<u64>()),
+        code: iter::repeat_with(random).take(AVERAGE_DATA_SIZE).collect(),
```
Comment (hrls) [inline evm-utils/evm-state/benches/squashed_state.rs]: randomize vec size also

File: evm-utils/evm-state/benches/squashed_state.rs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            BenchmarkId::from_parameter(layers),
+            &layers,
+            move |b, layers| {
+                b.iter_with_large_setup(
+                    || {
+                        let slot = Slot::default();
+                        let mut state = EvmState::default();
+
+                        let accounts = iter::repeat_with(|| (Address::random(), some_account()))
+                            .take(N_ACCOUNTS)
```
Comment (hrls) [inline evm-utils/evm-state/benches/squashed_state.rs]: configurable (accounts per slot)

File: evm-utils/evm-state/benches/squashed_state.rs [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                        state.squash();
+                    },
+                );
+            },
+        );
+    }
+
+    group.finish();
+
+    c.bench_function("squash_after_squash_100", |b| {
```
Comment (hrls) [inline evm-utils/evm-state/benches/squashed_state.rs]: merge with the previous one

**Review by hrls — COMMENTED** [review]
TODO

**Review by vldm — APPROVED** [review]

**Pull request merged**